decompy/RL/Model/Model_template.py

```diff
@@ -38,19 +38,61 @@ def get_past_info(self, summary_of_current_state, possible_actions):
             """
             pass
 
+        def get_decision_by_hash(self, hashed_llvm):
+            """
+            gets the decision by hashed_llvm summary.
+            :param hashed_llvm: the hashed_llvm summary
+            :return: a decision or None if not found.
+            :rtype: Decision
+            """
+            return self.decision_history[hashed_llvm]
+
+        def get_all_decisions_dictionary(self, hashed_llvm):
```

Comment:
why is your internal model a list if the main use case is to get the decision by it's summary? Are we going to reassemble the dictionary every time?